mlir/lib/Dialect/MQTRef/Translation/ImportQuantumComputation.cpp

```diff
@@ -0,0 +1,276 @@
+ * @param qubits The qubits of the quantum register.
+ */
+void addMeasureOp(mlir::OpBuilder& builder, const qc::Operation& operation,
+                  const llvm::SmallVector<mlir::Value>& qubits) {
+  const auto& bitType = mlir::IntegerType::get(builder.getContext(), 1);
+  const auto& targets = operation.getTargets();
+  for (const auto& target : targets) {
+    const mlir::Value inQubit = qubits[target];
+    builder.create<mqt::ir::ref::MeasureOp>(builder.getUnknownLoc(), bitType,
+                                            inQubit);
+  }
+}
+
+/**
+ * @brief Adds a reset operation to the MLIR module.
+ *
+ * @param builder The MLIR OpBuilder.
+ * @param operation The reset operation to add.
+ * @param qubits The qubits of the quantum register.
+ */
+void addResetOp(mlir::OpBuilder& builder, const qc::Operation& operation,
+                const llvm::SmallVector<mlir::Value>& qubits) {
+  const auto& targets = operation.getTargets();
+  for (const auto& target : targets) {
+    const mlir::Value inQubit = qubits[target];
+    builder.create<mqt::ir::ref::ResetOp>(builder.getUnknownLoc(), inQubit);
+  }
+}
+
+#define ADD_OP_CASE(op)                                                        \
+  case qc::OpType::op:                                                         \
+    addOperation<mqt::ir::ref::op##Op>(builder, *operation, qubits);           \
+    break;
+
+/**
+ * @brief Adds quantum operations to the MLIR module.
+ *
+ * @param builder The MLIR OpBuilder used to create operations
+ * @param quantumComputation The quantum computation to translate
+ * @param qubits The qubits of the quantum register
+ * @return mlir::LogicalResult Success if all operations were added, failure
+ * otherwise
+ */
+llvm::LogicalResult
+addOperations(mlir::OpBuilder& builder,
+              const qc::QuantumComputation& quantumComputation,
+              const llvm::SmallVector<mlir::Value>& qubits) {
+  for (const auto& operation : quantumComputation) {
+    switch (operation->getType()) {
+      ADD_OP_CASE(I)
+      ADD_OP_CASE(H)
+      ADD_OP_CASE(X)
+      ADD_OP_CASE(Y)
+      ADD_OP_CASE(Z)
+      ADD_OP_CASE(S)
+      ADD_OP_CASE(Sdg)
+      ADD_OP_CASE(T)
+      ADD_OP_CASE(Tdg)
+      ADD_OP_CASE(V)
+      ADD_OP_CASE(Vdg)
+      ADD_OP_CASE(U)
+      ADD_OP_CASE(U2)
+      ADD_OP_CASE(P)
+      ADD_OP_CASE(SX)
+      ADD_OP_CASE(SXdg)
+      ADD_OP_CASE(RX)
+      ADD_OP_CASE(RY)
+      ADD_OP_CASE(RZ)
+      ADD_OP_CASE(SWAP)
+      ADD_OP_CASE(iSWAP)
+      ADD_OP_CASE(iSWAPdg)
+      ADD_OP_CASE(Peres)
+      ADD_OP_CASE(Peresdg)
+      ADD_OP_CASE(DCX)
+      ADD_OP_CASE(ECR)
+      ADD_OP_CASE(RXX)
+      ADD_OP_CASE(RYY)
+      ADD_OP_CASE(RZZ)
+      ADD_OP_CASE(RZX)
+    case qc::OpType::XXminusYY:
+      addOperation<mqt::ir::ref::XXminusYY>(builder, *operation, qubits);
+      break;
+    case qc::OpType::XXplusYY:
+      addOperation<mqt::ir::ref::XXplusYY>(builder, *operation, qubits);
+      break;
```

Comment:
I am very sure I am overlooking things, but why are these two operations not covered by a similar macro?

Reply:
Certainly an oversight.
Might as well fix this here and add a separate Changelog entry.